src/Microsoft.SourceIndexer.Tasks/DownloadStage1Index.cs

```diff
@@ -118,14 +120,15 @@ private void ExecuteCore()
             try 
             {
                 using Stream fileStream = blobClient.OpenRead();
+                using var input = new GZipInputStream(fileStream);
+                using var archive = TarArchive.CreateInputTarArchive(input, Encoding.UTF8);
+                archive.ExtractContents(OutputDirectory, true); // would like this to be false, but SharpZipLib has a bug in 1.3.3
             }
             catch (AuthenticationFailedException e)
             {
-                Fatal($"*** STREAM READ FAILED: {e.Message}");
+                Log.LogError($"*** STREAM READ FAILED: {e.Message}");
+                return;
```

Comment:
This is the last line in this method, no need to return.
```suggestion
```

Reply:
You're not wrong, but semantically, I think it's safer to be explicit in case other code is added below at a later point (at which point the assumption about not needing return would no longer be true, and i'd be relying on future Jo to notice